framer/TextLayer.coffee

```diff
@@ -0,0 +1,123 @@
+{Layer} = require "./Layer"
+{Events} = require "./Events"
+
+class exports.TextLayer extends Layer
+
+	constructor: (options={}) ->
+
+		options.width ?= 300
+		options.backgroundColor ?= "transparent"
+		options.html ?= "Add text"
+		options.color ?= "#888"
+
+		super options
+
+		# Set type defaults
+		if not @fontFamily
+
+			currentDevice = Framer.Device.deviceType
+
+			# Apple Device: SF UI
+			if currentDevice.indexOf("apple") > -1
+				@_setStyle("-apple-system, SF UI Text, Helvetica Neue", @fontSize, @fontWeight, @lineHeight)
+
+			# Google Device: Roboto
+			if currentDevice.indexOf("google") > -1
+				@_setStyle("Roboto, Helvetica Neue", @fontSize, @fontWeight, @lineHeight)
+
+			if currentDevice.indexOf("microsoft") > -1
+				@_setStyle("Segoe UI, Helvetica Neue", @fontSize, @fontWeight, @lineHeight)
+
+			# Else: macOS
+			else
+				@_setStyle("-apple-system, SF UI Text, Helvetica Neue", @fontSize, @fontWeight, @lineHeight)
+
+		if not @fontSize
+			@_setStyle(@fontFamily, 40, @fontWeight, @lineHeight)
+
+		if not @fontWeight
+			@_setStyle(@fontFamily, @fontSize, 400, @lineHeight)
+
+		if not @lineHeight
+			@_setStyle(@fontFamily, @fontSize, @fontWeight, 1.25)
```

Comment:
Awesome, using the `@define` getters is much cleaner indeed. See: https://github.com/koenbok/Framer/pull/434/commits/b0526d6905a10b1a017a45e71b21233ef8b39aa0